sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -335,6 +335,12 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val ANSI_SQL_PARSER =
```

Comment:
@maropu Isn't this config too generic ? When this is set to true, can we confidently say that we follow ANSI standard in all other rules ? I don't know if DDLs are part of standard or not. If they are, then we may not be ansi compliant , right ?

Reply:
Sorry, but I might miss your point. Basically, we need to follow the SQL standard. So, we need to check if we could comply with the standard if the flag is true.

Reply:
@maropu OK.. may be i am misunderstanding. Basically i was thinking from an user's perspective. When i set this config, what would be my expectation from the system ?
I was asking, since we only make use of this config while parsing an interval value. Is the expectation that, we will eventually put more and more standard complaint features under this config ? So basically we are creating a generic config ahead in time.

Reply:
yea, yes. The config is not specific to this interval behaiour and I think we need to controll the all the parsing behaivour changes by this config for the standard.

Reply:
@maropu ok.. cool. I am now wondering about a config i created `spark.sql.legacy.setopsPrecedence.enabled` a while back . Should the set operation precedence be also controlled by this new config ? cc @gatorsmile for his input.